deepchem/hyper/grid_search.py

```diff
@@ -9,23 +7,39 @@
 import tempfile
 import shutil
 import collections
+import logging
 from functools import reduce
 from operator import mul
 from deepchem.utils.evaluate import Evaluator
-from deepchem.utils.save import log
+from deepchem.hyper.base_classes import HyperparamOpt
 
+logger = logging.getLogger(__name__)
 
-class HyperparamOpt(object):
-  """
-  Provides simple hyperparameter search capabilities.
+
+class GridHyperparamOpt(HyperparamOpt):
   """
+  Provides simple grid hyperparameter search capabilities.
+
+  This class performs a grid hyperparameter search over the specified
+  hyperparameter space. This implementation is simple and simply does
+  a direct iteration over all possible hyperparameters and doesn't use
+  parallelization to speed up the search.
+
+  Example
+  -------
+  This example shows the type of constructor function expected. 
 
-  def __init__(self, model_class, verbose=True):
-    self.model_class = model_class
-    self.verbose = verbose
+  >>> import sklearn
+  >>> import deepchem as dc
+  >>> def rf_model_builder(**model_params):
+  ...   rf_params = {k: v for (k, v) in model_params.items() if k != 'model_dir'}
+  ...   model_dir = model_params['model_dir']
+  ...   sklearn_model = sklearn.ensemble.RandomForestRegressor(**rf_params)
+  ...   return dc.models.SklearnModel(sklearn_model, model_dir)
```

Comment:
This may not be the best example.  For any standard DeepChem model the code would be much simpler, right?

```python
def gc_model_builder(**model_params):
  return dc.models.GraphConvModel(**model_params)
optimizer = dc.hyper.GridHyperparamOpt(gc_model_builder)
```

Or you could even skip defining the function and just write

```python
optimizer = dc.hyper.GridHyperparamOpt(lambda **p: dc.models.GraphConvModel(**p))
```

Reply:
This is a great suggestion! Much cleaner. Swapped over